brave/brave-core
# Add DNS-over-HTTPS P3A metrics

Resolves https://github.com/brave/brave-browser/issues/31965



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Ensure a DNS server that supports DoH (such as Cloudflare: 1.1.1.1) is configured in your OS network settings. Start with fresh profile.
1. Ensure that the secure setting metric has a value of 1.
2. Make a random query on your search engine, click on a bunch unique links.
3. On the local state page, search for `upgraded_dns_requests` and `total_dns_requests`. Note the values. `upgraded_dns_requests` should have a positive value, if your OS settings are correct.
4. The percentage range indicated by the value of the auto-secure requests metric should encompass the ratio of `upgraded_dns_requests` to `total_dns_requests` (i.e. if the upgraded count is 10, and the total count is 15, the ratio would be 66% and the correct metric answer would be 2)
5. Disable the DoH supporting server in your OS settings (let DHCP do it's thing)
6. Repeat step 2.
7. Ensure that the `upgraded_dns_requests` count does not grow as you visit websites. The ratio of upgraded-to-total requests should become lower as you visit more domains. The auto-secure requests metric value should also change accordingly.
8. Change your browser DoH setting to "With Cloudflare". `Brave.DNS.SecureSetting` should be 2, and `Brave.DNS.AutoSecureRequests` should disappear.
9. Turn off your browser DoH setting. Ensure both metrics do not exist.
